test/e2e/apimachinery/flowcontrol.go

```diff
@@ -113,11 +117,121 @@ var _ = SIGDescribe("[Feature:APIPriorityAndFairness] response header should pre
 		}
 	})
 
+	ginkgo.It("should ensure that requests can't be drowned out", func() {
+		flowSchemaNamePrefix := "e2e-testing-flowschema"
+		priorityLevelNamePrefix := "e2e-testing-prioritylevel"
+		loadDuration := 10 * time.Second
+		type client struct {
+			username                     string
+			qps                          float64
+			assuredConcurrencyShares     int32
+			matchingPrecedence           int32
+			completedRequests            int32
+			expectedMinCompletedRequests int32
+		}
+		clients := []client{
+			// An out-of-control elephant that not only has a larger concurrency
+			// share than its peers, but also misbehaves by trying to make requests
+			// at a much quicker rate than the mouse.
+			{username: "elephant", qps: 100.0, assuredConcurrencyShares: 20, matchingPrecedence: 999},
+			// A well behaved mouse that is making requests at a rate less than its
+			// assured concurrency shares. The elephant's flow schema has precedence
+			// over the mouse's.
+			{username: "mouse", qps: 5.0, assuredConcurrencyShares: 10, matchingPrecedence: 1000},
```

Comment:
i suppose the purposing e2e test in this pull is almost the same with the existing [integration test](https://github.com/kubernetes/kubernetes/blob/a69a4a4bf02248130b7aaacb74f031fe3faad485/test/integration/apiserver/flowcontrol/concurrency_test.go#L70-L146). additionally, i think we'd better not use a hardcoded QPS b/c our e2e suite is supposed to be always working regardless of the target server's performance/capacity? the integration test adaptively adjusts the clients' requesting rate by scraping actual shared concurrency from metrics to avoid flakiness.

Reply:
Done.